tests/test_s3_utils.py

```diff
@@ -46,7 +59,77 @@ def should_successfully_get_object_from_staging_bucket() -> None:
         key=f"{key_prefix}/{collection_metadata_filename}",
     ):
 
-        s3_url_reader = get_s3_url_reader(get_s3_role_arn())
-        json_object = load(s3_url_reader(collection_metadata_url))
+        s3_url_reader = get_s3_url_reader(get_s3_role_arn(), any_dataset_prefix(), get_log())
+        json_object = load(s3_url_reader(collection_metadata_url).response["Body"])
+
+        assert json_object == collection_dict
+
+
+@mark.infrastructure
+def should_get_object_from_storage_bucket_when_staging_bucket_fails() -> None:
+    key_prefix = any_safe_file_path()
+    metadata_url_prefix = (
+        f"{S3_URL_PREFIX}{Resource.STAGING_BUCKET_NAME.resource_name}/{key_prefix}"
+    )
+    collection_metadata_filename = any_safe_filename()
+    collection_metadata_url = f"{metadata_url_prefix}/{collection_metadata_filename}"
+
+    collection_dict = {
```

Comment:
It looks like the contents of this collection doesn't actually matter, so I think it could be simplified to just `deepcopy(MINIMAL_VALID_STAC_COLLECTION_OBJECT)` or something like `any_json_object()`.